UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Controllers/EssWebhookController.cs

```diff
@@ -45,23 +49,39 @@ public IActionResult NewFilesPublishedOptions()
         [Route("/webhook/newfilespublished")]
         public virtual async Task<IActionResult> NewFilesPublished([FromBody] JObject request)
         {
+            Logger.LogInformation(EventIds.ESSClearCacheSearchDownloadEventStart.ToEventId(), "Clear Cache Event started for _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+            var azureAdB2C = new AzureAdB2C()
+            {
+                AudToken = TokenAudience,
+                IssToken = TokenIssuer
+            };
+            bool isAzureB2C = azureAdB2CHelper.IsAzureB2CUser(azureAdB2C, GetCurrentCorrelationId());
+            if (isAzureB2C)
+            {
+                Logger.LogInformation(EventIds.ESSB2CUserValidationEvent.ToEventId(), "Event was triggered with invalid Azure AD token from Enterprise event for Clear Cache Search and Download Event for _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+                Logger.LogInformation(EventIds.ESSClearCacheSearchDownloadEventCompleted.ToEventId(), "Clear Cache Event completed as Azure AD Authentication failed with OK response and _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+                return GetCacheResponse();
+            }
+
             var eventGridEvent = JsonConvert.DeserializeObject<CustomEventGridEvent>(request.ToString());
```

Comment:
This seems to satisfy the CodeQL scanner and I don't see any problems making the change:
```suggestion
            var eventGridEvent = new CustomEventGridEvent();
            JsonConvert.PopulateObject(request.ToString(), eventGridEvent);
```
What do you think?

Reply:
Yes, It did worked. Thanks

Reply:
Thanks Barrie for looking into CodeQL scanning issue.